### PR TITLE
Fairygrass and bioluminescence port

### PR DESCRIPTION
**Pull request by slicerv**
# Document the changes in your pull request
Ports https://github.com/BeeStation/BeeStation-Hornet/pull/5411 and https://github.com/BeeStation/BeeStation-Hornet/pull/5654
Adds fairygrass from bee and also adds new bioluminescence colors to plants.
image

# Wiki Documentation
Add new grass types to botany wiki

# Changelog

:cl:  Pirill, Naevii, nianjiilical, Useroth, GregTheEgg42069, slicerv,
rscadd: Added Cherry Bulbs and fairygrass as new mutations to cherries and grass. 
rscadd:Pink,yellow,green.blue and purple bioluminescence traits to cherry bulbs, holymelons, omega weed, fairygrass and moonflowers 
rscadd: Added dark fairy grass
tweak: Shadowling veil changes the colored fairygrass to it's dark fairygrass.
/:cl:

### CODE REVIEW

**Review by ghost — APPROVED** [review]

**Comment by Simplehorror** [comment]
This will make strange seed traits quite bad. Strong bioluminescence is not only white but it has a higher range, please make it a seperate trait

**Comment by JamieD1** [comment]
Why..

**Comment by Simplehorror** [comment]
What would it look like if I released glowshrooms with varying colours?

File: code/modules/hydroponics/grown/grass_carpet.dm [hunk 1]
```diff
@@ -40,6 +40,57 @@
...
+				stacktype = /obj/item/stack/tile/fairygrass/blue
+			if(/datum/plant_gene/trait/glow/purple)
+				stacktype = /obj/item/stack/tile/fairygrass/purple
+			if(/datum/plant_gene/trait/glow/pink)
+				stacktype = /obj/item/stack/tile/fairygrass/pink
+			if(/datum/plant_gene/trait/glow/shadow)
+				stacktype = /obj/item/stack/tile/fairygrass/dark
+		
+
+	. = ..()
```
Comment (JamieD1) [inline code/modules/hydroponics/grown/grass_carpet.dm:92]:
```suggestion
		break
	stacktype = initial(stacktype)
	if(G)
		switch(G.type)
			if(/datum/plant_gene/trait/glow/white)
				stacktype = /obj/item/stack/tile/fairygrass/white
			if(/datum/plant_gene/trait/glow/red)
				stacktype = /obj/item/stack/tile/fairygrass/red
			if(/datum/plant_gene/trait/glow/yellow)
				stacktype = /obj/item/stack/tile/fairygrass/yellow
			if(/datum/plant_gene/trait/glow/green)
				stacktype = /obj/item/stack/tile/fairygrass/green
			if(/datum/plant_gene/trait/glow/blue)
				stacktype = /obj/item/stack/tile/fairygrass/blue
			if(/datum/plant_gene/trait/glow/purple)
				stacktype = /obj/item/stack/tile/fairygrass/purple
			if(/datum/plant_gene/trait/glow/pink)
				stacktype = /obj/item/stack/tile/fairygrass/pink
			if(/datum/plant_gene/trait/glow/shadow)
				stacktype = /obj/item/stack/tile/fairygrass/dark
	. = ..()
```

**Review by JamieD1 — APPROVED** [review]

**Comment by JamieD1** [comment]
Ok, so I tried to suggest changes and this showed an error with line endings, can you fix them please.

**Review by nesisknesis — APPROVED** [review]

**Comment by ChesterTheCheesy** [comment]
okay but am i the only person to think that grass tiles got replaced by an uglier sprite?

**Comment by Simplehorror** [comment]
I want to open a revert on this unless strong bio luminescence is added back, it wasn't in the changelog

**Comment by nmajask** [comment]
> I want to open a revert on this unless strong bio luminescence is added back, it wasn't in the changelog

If you want it back in, then just make a PR that adds it again instead of reverting a whole new feature to do it.

**Pull request merged**